sdks/java/core/src/test/java/org/apache/beam/sdk/io/FileSystemsTest.java

```diff
@@ -316,6 +317,24 @@ public void testInvalidSchemaMatchNewResource() {
     assertEquals("file", FileSystems.matchNewResource("c:/tmp/f1", false));
   }
 
+  @Test
```

Comment:
This unit test fails on windows. Will investigate further

Reply:
error on windows:

```
org.junit.ComparisonFailure: expected:<[/abc/d/]> but was:<[E:\abc\d\]>
	at org.junit.Assert.assertEquals(Assert.java:117)
```

Fixed by comparing resourceId and not converting to String, which will give OS specific valid paths